pkg/resources/file_format_grant.go

```diff
@@ -0,0 +1,206 @@
+		ForceNew:    true,
+	},
+	"database_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The name of the database containing the current or future file formats on which to grant privileges.",
+		ForceNew:    true,
+	},
+	"privilege": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		Description:  "The privilege to grant on the current or future file format.",
+		Default:      "USAGE",
+		ValidateFunc: validation.StringInSlice(validFileFormatPrivileges.toList(), true),
+		ForceNew:     true,
+	},
+	"roles": {
+		Type:        schema.TypeSet,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Grants privilege to these roles.",
+		ForceNew:    true,
+	},
+	"on_future": {
+		Type:          schema.TypeBool,
+		Optional:      true,
+		Description:   "When this is set to true and a schema_name is provided, apply this grant on all future file formats in the given schema. When this is true and no schema_name is provided apply this grant on all future file formats in the given database. The file_format_name field must be unset in order to use on_future.",
+		Default:       false,
+		ForceNew:      true,
+		ConflictsWith: []string{"file_format_name"},
+	},
+	"with_grant_option": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "When this is set to true, allows the recipient role to grant the privileges to other roles.",
+		Default:     false,
+		ForceNew:    true,
+	},
+}
+
+// FileFormatGrant returns a pointer to the resource representing a file format grant
+func FileFormatGrant() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateFileFormatGrant,
+		Read:   ReadFileFormatGrant,
+		Delete: DeleteFileFormatGrant,
+
+		Schema: fileFormatGrantSchema,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+// CreateFileFormatGrant implements schema.CreateFunc
+func CreateFileFormatGrant(data *schema.ResourceData, meta interface{}) error {
+	var (
+		fileFormatName string
+		schemaName     string
+	)
+	if _, ok := data.GetOk("file_format_name"); ok {
+		fileFormatName = data.Get("file_format_name").(string)
+	}
+	if _, ok := data.GetOk("schema_name"); ok {
+		schemaName = data.Get("schema_name").(string)
+	}
+	dbName := data.Get("database_name").(string)
+	priv := data.Get("privilege").(string)
+	futureFileFormats := data.Get("on_future").(bool)
+	grantOption := data.Get("with_grant_option").(bool)
+
+	if (schemaName == "") && !futureFileFormats {
+		return errors.New("schema_name must be set unless on_future is true.")
+	}
+
+	if (fileFormatName == "") && !futureFileFormats {
+		return errors.New("file_format_name must be set unless on_future is true.")
+	}
+	if (fileFormatName != "") && futureFileFormats {
+		return errors.New("file_format_name must be empty if on_future is true.")
+	}
+
```

Comment:
As with above, it would be great to pull these up to the schema level.

Reply:
Done